// viem/dist/cjs/chains/definitions/base.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.base = void 0;
const chain_js_1 = require("../../utils/chain.js");
const fees_js_1 = require("../optimism/fees.js");
const formatters_js_1 = require("../optimism/formatters.js");
exports.base = (0, chain_js_1.defineChain)({
    id: 8453,
    network: 'base',
    name: 'Base',
    nativeCurrency: { name: 'Base', symbol: 'ETH', decimals: 18 },
    rpcUrls: {
        default: {
            http: ['https://mainnet.base.org'],
        },
        public: {
            http: ['https://mainnet.base.org'],
        },
    },
    blockExplorers: {
        blockscout: {
            name: 'Basescout',
            url: 'https://base.blockscout.com',
        },
        default: {
            name: 'Basescan',
            url: 'https://basescan.org',
        },
        etherscan: {
            name: 'Basescan',
            url: 'https://basescan.org',
        },
    },
    contracts: {
        multicall3: {
            address: '0xca11bde05977b3631167028862be2a173976ca11',
            blockCreated: 5022,
        },
    },
}, {
    fees: fees_js_1.feesOptimism,
    formatters: formatters_js_1.formattersOptimism,
});
