// viem/chains/dist/cjs/utils.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.formattersOptimism = exports.parseTransactionCelo = exports.serializersCelo = exports.serializeTransactionCelo = exports.formattersCelo = exports.defineChain = void 0;
function defineChain(chain, config = {}) {
    const { fees = chain.fees, formatters = chain.formatters, serializers = chain.serializers, } = config;
    return {
        ...chain,
        fees,
        formatters,
        serializers,
    };
}
exports.defineChain = defineChain;
var formatters_js_1 = require("./celo/formatters.js");
Object.defineProperty(exports, "formattersCelo", { enumerable: true, get: function () { return formatters_js_1.formattersCelo; } });
var serializers_js_1 = require("./celo/serializers.js");
Object.defineProperty(exports, "serializeTransactionCelo", { enumerable: true, get: function () { return serializers_js_1.serializeTransactionCelo; } });
Object.defineProperty(exports, "serializersCelo", { enumerable: true, get: function () { return serializers_js_1.serializersCelo; } });
var parsers_js_1 = require("./celo/parsers.js");
Object.defineProperty(exports, "parseTransactionCelo", { enumerable: true, get: function () { return parsers_js_1.parseTransactionCelo; } });
var formatters_js_2 = require("./optimism/formatters.js");
Object.defineProperty(exports, "formattersOptimism", { enumerable: true, get: function () { return formatters_js_2.formattersOptimism; } });
